app/code/Magento/Catalog/Model/ResourceModel/Category/Flat/Collection.php

```diff
@@ -125,11 +122,9 @@ public function addIdFilter($categoryIds)
             $condition = $categoryIds;
         } elseif (is_string($categoryIds)) {
             $ids = explode(',', $categoryIds);
-            if (empty($ids)) {
-                $condition = $categoryIds;
-            } else {
-                $condition = ['in' => $ids];
-            }
+            $condition = ['in' => $ids];
```

Comment:
This seems to be a change that could influence the method. Please add it in the description of the pr.

Reply:
The result of `explode()` will never be `empty()`, so the change here is removing dead code. There is no change to functionality here, as the code would never have been reached.

https://3v4l.org/B7NTo
![Screenshot_2023-05-29_21-30-25](https://github.com/magento/magento2/assets/334786/3a69afcf-787f-4331-b405-ce31b992c118)